### PR TITLE
Relationship strength

### PR DESCRIPTION
**Pull request by szhu**

### CODE REVIEW

File: source/includes/_relationship_strengths.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+If an internal and external person have no previous interactions, there may be no relation
+strength resource for their relationship.
+
+## The relationship strength resource
+
+> Example Response
+
+```json
+{
+  "external_person_id": 1234,
```
Comment (Hansenq) [inline source/includes/_relationship_strengths.md]: tbh, maybe we should rename these `external_id` and `internal_id`...

File: source/includes/_relationship_strengths.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+```
+
+Attribute | Type | Description
+--------- | ------- | -----------
+internal_person_id | integer | The internal person associated with this relationship strength.
+external_person_id | integer | The external person associated with this relationship strength.
+strength | float | The actual relationship strength. This is currently a number between 0 and 1, but may change in the future.
+
+
+## Search for a relationship strength
```
Comment (Hansenq) [inline source/includes/_relationship_strengths.md]: This isn't really a search; it's more like a fetch

File: source/includes/_relationship_strengths.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+```shell
+# Returns an array relationship strengths matching the criteria.
+curl "https://api.affinity.co/relationships-strengths?external_person_id=1234&internal_person_id=2345" -u :<API-KEY>
+```
+
+> Example Response
+
+```json
+{
+  "relationship_strengths": [
```
Comment (Hansenq) [inline source/includes/_relationship_strengths.md]: Looking at your PR, the response is not
```json
{
  "relationship_strengths": [
    {
      "internal_person_id": 1234,
      ...
    },
    ...
  ],
}
```
but actually
```json
[
  {
    "internal_person_id": 1234,
    ...
  },
  ...
]
```

File: source/includes/_relationship_strengths.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+```shell
+# Returns an array relationship strengths matching the criteria.
+curl "https://api.affinity.co/relationships-strengths?external_person_id=1234" -u :<API-KEY>
+```
+
+> Example Response
+
+```json
+{
+  "relationship_strengths": [
```
Comment (Hansenq) [inline source/includes/_relationship_strengths.md]: Same here

File: source/includes/_relationship_strengths.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+}
+```
+
+Get the relationship strength between all internal people and an external person:
+
+> Example Request
+
+```shell
+# Returns an array relationship strengths matching the criteria.
+curl "https://api.affinity.co/relationships-strengths?external_person_id=1234" -u :<API-KEY>
```
Comment (Hansenq) [inline source/includes/_relationship_strengths.md]: We should mention how the results are sorted, if they are at all.

(looking at your PR they're sorted by internal person ID? Should they be sorted on strength instead?)

File: source/includes/_relationship_strengths.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
+# Relationship Strengths
+
+Affinity calculates relationship strengths between internal and external people based on
+previous interactions (emails, logged calls, calendar events).
```
Comment (Hansenq) [inline source/includes/_relationship_strengths.md]: I think it'd be useful to give the users a bit more information about how this resource is generated and what they can assume from this resource. Something to tell them that:
- In general, a higher numeric value means that the relationship strength between the two people is higher, but may be inaccurate
- These values are generally refreshed daily
- Internal/external corresponds to the Person type (as defined above)
- There may be at most one resource for every (internal, external) pair

File: source/includes/_relationship_strengths.md (hunk 5 above)
Comment (szhu) [reply source/includes/_relationship_strengths.md]: Here's my reasoning behind leaving that part out:

> I'd rather pick a meaningless but stable default sort like internal person ID. We can add in a sort parameter later if they ask for it. The advantage of having a sort parameter is that we can see how customers want this data sorted.

File: source/includes/_relationship_strengths.md (hunk 1 above)
Comment (szhu) [reply source/includes/_relationship_strengths.md]: I lean towards not doing that, for consistency purposes. (The long name is consistent with what we use internally, and is very clear about the type of object (Person) that the Id comes from.)

How do you weigh these pros and cons?

File: source/includes/_relationship_strengths.md (hunk 6 above)
Comment (szhu) [reply source/includes/_relationship_strengths.md]: Ah, this is good info!

Fixed in 4752fbb574bfc7cf6a846b1ff7777b85eb796e3b.

File: source/includes/_relationship_strengths.md (hunk 2 above)
Comment (szhu) [reply source/includes/_relationship_strengths.md]: Thanks! Fixed in 02da00cec5d02feab60d067ef2f32c63f9086f85

File: source/includes/_relationship_strengths.md (hunk 3 above)
Comment (szhu) [reply source/includes/_relationship_strengths.md]: Thanks for catching this! I thought I remembered what the format was but I was clearly mistaken.

Fixed by 57341e988d928ba1e66259c4b6c2c42a73310c17
I also double-checked by running the API server, and this is now the correct format.

File: source/includes/_relationship_strengths.md (hunk 4 above)
Comment (szhu) [reply source/includes/_relationship_strengths.md]: Fixed by 57341e988d928ba1e66259c4b6c2c42a73310c17

File: source/includes/_relationship_strengths.md (hunk 5 above)
Comment (szhu) [reply source/includes/_relationship_strengths.md]: Resolved offline. Fixed in b4e3c1160f713159eb795073fb097fb019fdeaa6

File: source/includes/_relationship_strengths.md (hunk 1 above)
Comment (szhu) [reply source/includes/_relationship_strengths.md]: Done!
https://github.com/affinity-team/affinity/pull/3513
Fixed in 07d53b349158b4e9d903f40105b73ab081a94482

**Pull request merged**